_drafts/partially-ordering-time.md

```diff
@@ -0,0 +1,253 @@
+supposed to, anyway; in practice many hardware memory models are
+[underspecified](#urls[1-2]).)
+
+Your compiler will have a memory model; this captures how your language chooses to expose
+the hardware memory model to you, as well as what license your compiler takes in
+reordering instructions _while generating assembly_ across multiple threads. This is what
+you code to when writing lock-free systems code.
+
+In practice, there's a single weak memory model that matters to most people—the C++11
+memory model, as developed to be an abstraction over all of x86, POWER, and ARM.
+
+And your distributed system will have a consistency model; this captures how your system
+chooses to expose the synchronisation or lack thereof in the system to you, as well as
+what “license” your network “takes” in reordering events across clients in a wide-area
+network. This is what you code to when writing a distributed application.
+
+In practice, there's a vast [consistency zoo](#url) of consistency models you could be
+using when writing a distributed system. Good. Good. Good!
+
+In all these cases, these models describe the (desired) observable behaviour of the
+system, from outside the system. If I, a single client, or a single thread, write to a
+value, and then immediately read from it, am I guaranteed to see a write at least as new
+as mine? If time wasn't weird, if we didn't constantly take license to reorder operations
+in all of these systems, the answer would be yes, of course.
+
+But time is weird, and so we have to ask.
+
+### Consistency Models—I Mean, Memory Models
+
+Both these worlds started out by trying to ignore time being this weird for as long as
+they could—distributed programmers tried to hue to ACID properties as long as possible,
+and systems programmers relied on mutual exclusion / locks for a long time.
+
+But we all need speed. Whether it's the CAP theorem and weakly/eventually consistent
+models [#url], or lockfree programming with weak/relaxed memory models, both worlds ended
+up relaxing this strong view of time in order to gain speed.
+
+Weak memory models and consistency models are both _abstract_ models: they describe the
+_interface_ of the system to the programmer using it. They describe what sorts of
+behaviours we _can_ rely on from time, now that the default properties we tend to assume
+no longer apply.
+
+Both worlds have developed their own language for describing these models. For instance:
+the phrase “data race” in systems programming refers to any pair of accesses to a memory
+location that include at least one write, without a synchronisation barrier between
+them. This concept doesn't have a specific phrasing in distributed programming, because
+that field assumes that reads and writes participate in data races (“are concurrent”) by
+default.
+
+Thus, “data-race freedom” is a stronger condition than the distributed systems term
+“sequential consistency” (“any concurrent accesses can be treated _as if_ they were
```

Comment:
I don't think quoting the term and the intuition in parentheses is a good idea. Kinda seems like you're defining a term with something that isn't defined?